src/main/java/taxi/dao/jdbcimpl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,133 @@
+package taxi.dao.jdbcimpl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import taxi.dao.ManufacturerDao;
+import taxi.exception.DataProcessingException;
+import taxi.lib.Dao;
+import taxi.model.Manufacturer;
+import taxi.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturers "
+                + "(manufacturer_name, manufacturer_country)"
+                + " VALUES(?, ?);";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                    PreparedStatement preparedStatement = connection
+                        .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            resultSet.close();
```

Comment:
it's a bit redundant to close result set. If statement is closed correctly, resultSet will be closed as well
https://docs.oracle.com/en/java/javase/11/docs/api/java.sql/java/sql/Statement.html#close()

Reply:
Removed, reviewed the article - now it's clear to me. Thank you